SensorArray.cpp

```diff
@@ -0,0 +1,94 @@
+#include "SensorArray.h"
+#include "Sensor.h"
+
+
+using namespace std;
+
+namespace LFRobot
+{
+	SensorArray::SensorArray(int pins[])
+	{
+		Sensor* sensorArray[NUMBER_OF_SENSORS];
+
+		for (int i = 0; i < NUMBER_OF_SENSORS; i++)
+		{
+			sensorArray[i] = new Sensor(pins[i]);
+		}
+	}
+
+	SensorArray::~SensorArray()
+	{
+		for (int i = 0; i < NUMBER_OF_SENSORS; i++)
+		{
+			delete sensorArray[i];
```

Comment:
If `sensorArray` were to be properly initialized (see above) then this is the correct way to handle the de-allocations 🥇